test/k8sT/Updates.go

```diff
@@ -232,6 +232,44 @@ func InstallAndValidateCiliumUpgrades(kubectl *helpers.Kubectl, oldHelmChartVers
 
 	testfunc := func() {
 		By("Deleting Cilium and CoreDNS...")
+
+		// Create a persistent connection to test that it persists
+		// across cilium reloads.
+		persistCtx, persistCancel := context.WithCancel(context.Background())
+		appPods := helpers.GetAppPods(apps, helpers.DefaultNamespace, kubectl, "id")
+		pod, ok := appPods[helpers.App2]
+		Expect(ok, BeTrue(), fmt.Sprintf("pod for %q app not found", helpers.App2))
+		persistConnCmd := kubectl.ExecPodCmdBackground(persistCtx,
+			helpers.DefaultNamespace, pod,
+			helpers.InfinitePing(app1Service))
+
+		defer func() {
+			persistCancel()
+			persistConnCmd.WaitUntilFinish()
+			ec := persistConnCmd.GetExitCode()
+			Expect(ec).Should(Equal(0), "persistent connection died in reload")
+			buf := persistConnCmd.CombineOutput()
+			out := buf.String()
+			errIdx := strings.Index(out, "errors,")
+			if errIdx == -1 {
+				return
+			}
+			// the format of a ping summary is something like:
+			// 10 packets transmitted, 7 received, +3 errors, 30% packet loss, time 9063ms
+			// look for the plus sign so we can scan the string
+			for ; ; errIdx-- {
+				if out[errIdx] == '+' {
+					break
+				}
+			}
+			var errCount int
+			var packetLoss float64
+			_, err := fmt.Sscanf(out[errIdx:], "%d errors, %f%% packet loss", &errCount, &packetLoss)
+			Expect(err).Should(BeNil(), "Unable to parse ping result")
+			Expect(errCount).Should(Equal(0), "ping had errors")
+			Expect(packetLoss).Should(Equal(0.0), "ping had packet loss")
```

Comment:
These should be `ExpectWithOffset` because it's inside a function. And in this case, we are inside two functions, so it should be an offset of 2.